src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/Questionnaire.kt

```diff
@@ -0,0 +1,82 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa
+
+import jakarta.persistence.CascadeType
+import jakarta.persistence.Column
+import jakarta.persistence.Convert
+import jakarta.persistence.Entity
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.OneToMany
+import jakarta.persistence.SequenceGenerator
+import jakarta.persistence.Table
+import org.hibernate.Hibernate
+import org.hibernate.annotations.Generated
+import org.hibernate.type.YesNoConverter
+import org.springframework.data.annotation.LastModifiedBy
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@Entity
+@Table(name = "QUESTIONNAIRES")
+data class Questionnaire(
+  @Id
+  @Column(name = "QUESTIONNAIRE_ID")
+  @SequenceGenerator(name = "QUESTIONNAIRE_ID", sequenceName = "QUESTIONNAIRE_ID", allocationSize = 1)
+  @GeneratedValue(generator = "QUESTIONNAIRE_ID")
+  var id: Long = 0,
+
+  @Column(name = "DESCRIPTION")
+  val description: String? = null,
+
+  @Column
+  var code: String,
+
+  @Column(name = "ACTIVE_FLAG", nullable = false)
+  @Convert(converter = YesNoConverter::class)
+  val active: Boolean = true,
+
+  @Column(name = "QUESTIONNAIRE_CATEGORY", nullable = false)
+  val category: String = "IR_TYPE",
+
+  @Column(name = "LIST_SEQ", nullable = false)
+  val listSequence: Int,
+
+  @OneToMany(cascade = [CascadeType.ALL], orphanRemoval = true)
+  @JoinColumn(name = "QUESTIONNAIRE_ID", nullable = false)
+  val questions: MutableList<QuestionnaireQuestion> = mutableListOf(),
+
+  @Column
+  var auditModuleName: String? = null,
+
+  @Column(name = "EXPIRY_DATE")
+  val expiryDate: LocalDate? = null,
+
+  @Column(name = "CREATE_USER_ID", nullable = false, insertable = false, updatable = false)
```

Comment:
See CourtEvent etc for a slightly better way as defining these as "generated" - remove them from constructor